server/finder/handler/finder_handler.go

```diff
@@ -163,13 +163,55 @@ func (h *FinderHandler) ListProviders() ([]byte, error) {
 				log.Errorw("Could not get provider index count", "err", err)
 			}
 		}
-		responses[i] = model.MakeProviderInfo(infos[i].AddrInfo, infos[i].LastAdvertisement,
-			infos[i].LastAdvertisementTime, infos[i].Publisher, infos[i].PublisherAddr, indexCount)
+		pInfo := infos[i]
+		responses[i] = model.MakeProviderInfo(pInfo.AddrInfo, pInfo.LastAdvertisement,
+			pInfo.LastAdvertisementTime, pInfo.Publisher, pInfo.PublisherAddr, indexCount)
+
+		responses[i].ExtendedProviders = makeExtendedProviders(pInfo)
 	}
 
 	return json.Marshal(responses)
 }
 
+func makeExtendedProviders(pInfo *registry.ProviderInfo) *model.ExtendedProviders {
```

Comment:
this function is not in `provider_info.go` because that would introduce a circular import